promql/functions.go

```diff
@@ -784,8 +784,42 @@ func funcMadOverTime(vals []parser.Value, args parser.Expressions, enh *EvalNode
 	}), annos
 }
 
+// === ts_of_last_over_time(Matrix parser.ValueTypeMatrix) (Vector, Notes)  ===
+func funcTsOfLastOverTime(vals []parser.Value, _ parser.Expressions, enh *EvalNodeHelper) (Vector, annotations.Annotations) {
+	el := vals[0].(Matrix)[0]
+
+	var tf float64
+	if len(el.Floats) > 0 {
+		tf = float64(el.Floats[len(el.Floats)-1].T / 1000)
+	}
+
+	var th float64
+	if len(el.Histograms) > 0 {
+		th = float64(el.Floats[len(el.Floats)-1].T / 1000)
+	}
+
+	return append(enh.Out, Sample{
+		Metric: el.Metric,
+		F:      max(tf, th),
```

Comment:
Maybe it's cleaner to make both `tf` and `th` int64, don't divide by 1000 above, and then do
```suggestion
		F:      float64(max(tf, th))/1000,
```
here.

Reply:
nice catch!